packages/webviz-core/src/panels/ThreeDimensionalViz/commands/utils/index.js

```diff
@@ -36,76 +36,113 @@ export function setRgba(buffer: Uint8Array, index: number, color: tinycolor) {
+  buff[buff_index++] = 0x70;  // red
+  buff[buff_index++] = 0x89;  // green
+  buff[buff_index++] = 0x86;  // blue
+  buff[buff_index++] = 255;   // alpha
 
   return buff;
 })();
 
 
+// Ported from Rviz's Costmap palette
+// https://github.com/ros-visualization/rviz/blob/22b81ecfea7ea7ed69e35d537abf6f50c8e5f1d7/src/rviz/default_plugin/map_display.cpp#L322
 export const defaultObstacleGridPalette = (() => {
   const buff = new Uint8Array(256 * 4);
-
-  // standard gray map palette values
-  for (let i = 0; i <= 100; i++) {
-    const t = 1 - i / 100;
-    const idx = i * 4;
-    setRgba(buff, idx, tinycolor.fromRatio({ r: t, g: t, b: t }));
+  let buff_index = 0;
+
+  // zero values have alpha=0
+  buff[buff_index++] = 0;  // red
+  buff[buff_index++] = 0;  // green
+  buff[buff_index++] = 0;  // blue
+  buff[buff_index++] = 0;  // alpha
+
+  // Blue to red spectrum for most normal cost values
+  for (let i = 1; i <= 98; i++)
+  {
+    const v = (255 * i) / 100;
+    buff[buff_index++] = v;        // red
+    buff[buff_index++] = 0;        // green
+    buff[buff_index++] = 255 - v;  // blue
+    buff[buff_index++] = 255;      // alpha
   }
 
+  // inscribed obstacle values (99) in cyan
+  buff[buff_index++] = 0;     // red
+  buff[buff_index++] = 255;  // green
+  buff[buff_index++] = 255;  // blue
+  buff[buff_index++] = 255;  // alpha
+
+  // lethal obstacle values (100) in purple
+  buff[buff_index++] = 255;  // red
+  buff[buff_index++] = 0;    // green
+  buff[buff_index++] = 255;  // blue
+  buff[buff_index++] = 255;  // alpha
+
   // illegal positive values in green
-  for (let i = 101; i <= 127; i++) {
-    const idx = i * 4;
-    setRgba(buff, idx, tinycolor("lime"));
+  for (let i = 101; i <= 127; i++)
+  {
+    buff[buff_index++] = 0;    // red
+    buff[buff_index++] = 255;  // green
+    buff[buff_index++] = 0;    // blue
+    buff[buff_index++] = 255;  // alpha
   }
 
-  // illegal negative (char) values
-  for (let i = 128; i <= 248; i++) {
-    const idx = i * 4;
-    const t = (i - 128) / (254 - 128);
-    setRgba(buff, idx, tinycolor.fromRatio({ r: t, g: 0.2, b: 0.6, a: Math.max(1 - t, 0.2) }));
+  // illegal negative (char) values in shades of red/yellow
+  for (let i = 128; i <= 254; i++)
+  {
+    buff[buff_index++] = 255;                              // red
+    buff[buff_index++] = (255 * (i - 128)) / (254 - 128);  // green
+    buff[buff_index++] = 0;                                // blue
+    buff[buff_index++] = 255;                              // alpha
   }
 
-  // legal negative values
+  // legal -1 value is tasteful blueish greenish grayish color
+  buff[buff_index++] = 0x70;  // red
+  buff[buff_index++] = 0x89;  // green
+  buff[buff_index++] = 0x86;  // blue
+  buff[buff_index++] = 255;   // alpha
+
+  // Override individual colors below
   // https://github.com/cjds/carl/blob/master/carrack_ros/map_updater/include/impl/controller_costmap_republisher_block.hpp#L40
```

Comment:
Probs should remove this comment. This is no longer copied from here

Reply:
Maybe leave a comment to that effect?

Reply:
Done